### ISSUE TITLE
Select Rows - do not change filter after column change if it is not required

### ISSUE THREAD

**Issue by Hrovatin** [biolab/orange3#4737]
When performing filtering with Select Rows the filters are reset after changing the column on which the filtering is performed. This is annoying when one has multiple similar columns that require the same filters. Could there be an option not to reset the filter columns, except if necessary due to a variable type change?
Additionally, even if a filter is changed from colA to colB and back to colA the previous settings used for colA are not retained.

Version:
Orange 3.26.

**Comment by janezd** [comment]
> When performing filtering with Select Rows the filters are reset after changing the column on which the filtering is performed.

Indeed, it requires all variables. It suppose it would be possible to modify the context handler by relaxing the match method and adding `filter_values` to remove non-matching conditions.

> Additionally, even if a filter is changed from colA to colB and back to colA the previous settings used for colA are not retained.

This happens on your request. :) Some time ago we modified the behaviour of contexts to match the most recent context. This is not fixable (= not compatible with the current strategy), but may no longer matter after/if we fix the first point.

**Comment by Hrovatin** [comment]
Yes, the second comment was just because I remembered that we previously discussed that Orange remembers contexts and was not sure if my main request would be irrelevant because of it. Maybe Orange could have options in widgets to set how to remember things? (If that would not be too cluttering.) The best remembering option is highly context dependent.

**Comment by janezd** [comment]
Actually, #4740 might fulfill all your wishes (regarding the Select Rows widget, that is). Try it.

I wouldn't let the user select the way contexts are matched because few people would understand what it does.

**Comment by Hrovatin** [comment]
The #4740  did not change the behaviour.

**Comment by janezd** [comment]
Hm, it worked for me. Have you tried with clear settings?

**Comment by Hrovatin** [comment]
Yes, but still the old behaviour. I tried on my workflow and on Iris for first setting one filter to _is outside_ and then changing the numerical column which set the filter back to _equals_.

**Comment by ajdapretnar** [comment]
I can confirm. The dropdown changes to default.

**Comment by janezd** [comment]
What am I missing? What am I doing differently?

![select-rows](https://user-images.githubusercontent.com/2387315/81225664-e0ce7480-8fe9-11ea-91b9-ce321b873a38.gif)

**Comment by Hrovatin** [comment]
That is not what I meant. The problem arises if you consider only the topmost filter: 
First you have: sepal length, is outside, [2,4].
Then you change this first filter to sepal width (leaving other two parameters intact). Expected: sepal width, is outside, [2,4]. Actual: sepal width, equals, 2. 
So the second and third column change, although only the first column was changed manually. Since both original and new variable are of the same type the change of the second and third column is not necessary. I would expect the other two columns to remain unchanged (or be set to whatever was previously used for the new variable (sepal width), if this is the Orange way of doing things). Although in my use case I would strongly prefer if second and third column remain as they are. Namely, I had problems in past when things changed by themselves and I did not notice it.
